src/main/routes/receiver.ts

```diff
@@ -1,49 +1,192 @@
+  if (roles.length === 1) {
+    return roles.pop().replace('letter-', '')
+  }
+
+  throw new Error('User was logged in but didn’t have a letter holder role')
+}
+
+function loginErrorHandler (req: express.Request, res: express.Response, next: express.NextFunction, err: Error) {
+  if (hasTokenExpired(err)) {
+    res.clearCookie(sessionCookie)
+    logger.debug(`Protected path - expired auth token - access to ${req.path} rejected`)
+    return res.redirect(authenticationRedirect.forLogin(req, res))
+  }
+  return next(err)
+}
 
 export default express.Router()
-  .get(AppPaths.receiver.uri, ErrorHandling.apply(async (req: express.Request, res: express.Response): Promise<void> => {
-    if (req.query.jwt) {
-      const sessionCookie = config.get<string>('session.cookieName')
-      new Cookies(req, res).set(sessionCookie, req.query.jwt, { sameSite: 'lax' })
-    }
-
-    const user: User = res.locals.user
-    const atLeastOneClaimIssued: boolean = (await ClaimStoreClient.retrieveByClaimantId(user.id)).length > 0
-    const claimAgainstDefendant = await ClaimStoreClient.retrieveByDefendantId(user.id)
-    const atLeastOneResponse: boolean = claimAgainstDefendant.length > 0 &&
-      claimAgainstDefendant.some((claim: Claim) => !!claim.response)
-
-    if (atLeastOneClaimIssued || atLeastOneResponse) {
-      return res.redirect(DashboardPaths.dashboardPage.uri)
-    }
-    const draftClaimSaved: boolean = user.claimDraft.document && user.claimDraft.id !== undefined
-    const claimIssuedButNoResponse: boolean = (claimAgainstDefendant).length > 0
-      && !atLeastOneResponse
-
-    if (claimIssuedButNoResponse && draftClaimSaved) {
-      return res.redirect(DashboardPaths.dashboardPage.uri)
-    }
-
-    if (draftClaimSaved) {
-      return res.redirect(ClaimPaths.taskListPage.uri)
-    }
-
-    if (claimIssuedButNoResponse) {
-      return res.redirect(ResponsePaths.taskListPage
-        .evaluateUri({ externalId: claimAgainstDefendant.pop().externalId }))
-    }
-
-    return res.redirect(ClaimPaths.startPage.uri)
-  }))
+  .get(AppPaths.receiver.uri,
+    ErrorHandling.apply(async (req: express.Request,
+                               res: express.Response,
+                               next: express.NextFunction): Promise<void> => {
+      const cookies = new Cookies(req, res)
+
+      const authenticationToken = await getAuthToken(req)
+      if (authenticationToken) {
+        const user = await IdamClient
+          .retrieveUserFor(authenticationToken)
+          .catch(err => loginErrorHandler(req, res, next, err))
+        if (!user) {
+          return
+        }
+        res.locals.isLoggedIn = true
+        res.locals.user = user
+        cookies.set(sessionCookie, authenticationToken, { sameSite: 'lax' })
+      }
+
+      if (isDefendantFirstContactPinLogin(req)) {
+        return res.redirect(FirstContactPaths.claimSummaryPage.uri)
+      }
+
+      if (res.locals.isLoggedIn) {
+        const user: User = res.locals.user
```

Comment:
Could we extract the content of this if block to e.g. `redirectToCorrespondingLandingPage`?